frontend/src/pages/modelServing/screens/metrics/MetricsPageToolbar.tsx

```diff
@@ -1,28 +1,30 @@
 import * as React from 'react';
-import {
-  Button,
-  Select,
-  SelectOption,
-  Text,
-  Toolbar,
-  ToolbarContent,
-  ToolbarItem,
-} from '@patternfly/react-core';
-import { SyncAltIcon } from '@patternfly/react-icons';
-import { TimeframeTitle } from '~/pages/modelServing/screens/types';
-import { relativeTime } from '~/utilities/time';
-import { isTimeframeTitle } from './utils';
+import { Select, SelectOption, Toolbar, ToolbarContent, ToolbarItem } from '@patternfly/react-core';
+import { RefreshIntervalTitle, TimeframeTitle } from '~/pages/modelServing/screens/types';
+import { isRefreshIntervalTitle, isTimeframeTitle } from './utils';
 import { ModelServingMetricsContext } from './ModelServingMetricsContext';
 
-const MetricsPageToolbar: React.FC = () => {
+type MetricsPageToolbarProps = {
+  leftToolbarItem?: React.ReactElement<typeof ToolbarItem>;
+};
+
+const MetricsPageToolbar: React.FC<MetricsPageToolbarProps> = ({ leftToolbarItem }) => {
   const [timeframeOpen, setTimeframeOpen] = React.useState(false);
-  const { currentTimeframe, setCurrentTimeframe, refresh, lastUpdateTime } = React.useContext(
-    ModelServingMetricsContext,
-  );
+  const {
+    currentTimeframe,
+    setCurrentTimeframe,
+    currentRefreshInterval,
+    setCurrentRefreshInterval,
+  } = React.useContext(ModelServingMetricsContext);
+
+  const [intervalOpen, setIntervalOpen] = React.useState(false);
+
   return (
     <Toolbar>
       <ToolbarContent>
-        <ToolbarItem>
+        {leftToolbarItem && leftToolbarItem}
+        <ToolbarItem alignment={{ default: 'alignRight' }}>
```

Comment:
Also, maybe consider change this to `ToolbarGroup` and stack the items inside it and give it a gutter.